bundles/org.openhab.binding.emby/src/main/java/org/openhab/binding/emby/internal/handler/EmbyBridgeHandler.java

```diff
@@ -0,0 +1,307 @@
+                }, CONNECTION_CHECK_INTERVAL_MS, CONNECTION_CHECK_INTERVAL_MS, TimeUnit.MILLISECONDS);
+
+            } catch (Exception e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Connection attempt failed: " + e.getMessage());
+            }
+        });
+    }
+
+    @Override
+    public void initialize() {
+        // Indicate that we're trying to connect
+        updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, "Attempting to connect to Emby server");
+
+        // Background setup
+        scheduler.execute(() -> {
+            this.reconnectionCount = 0;
+            this.config = new EmbyBridgeConfiguration();
+            if (callbackIpAddress != null) {
+                logger.debug("The callback ip address is: {}", callbackIpAddress);
+            } else {
+                logger.warn("Callback IP address is not set");
+            }
+            if (config.api != null && !config.api.isEmpty()) {
+                this.httputils = new EmbyHTTPUtils(30, config.api, getServerAddress());
+            }
+            this.connection = new EmbyConnection(this, this.webSocketClient);
+            try {
+                EmbyBridgeConfiguration cfg = checkConfiguration();
+                this.config = cfg;
+                establishConnection();
+            } catch (ConfigValidationException cve) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Configuration error: " + cve.getMessage());
+            }
+        });
+    }
+
+    @Override
+    public void handleEvent(EmbyPlayStateModel playstate, String hostname, int embyport) {
+        EmbyClientDiscoveryService discoveryService = this.clientDiscoveryService;
+        if ((discoveryService != null) && config.discovery) {
+            discoveryService.addDeviceIDDiscover(playstate);
+        }
+        this.getThing().getThings().forEach(thing -> {
+            EmbyDeviceHandler handler = (EmbyDeviceHandler) thing.getHandler();
+            if (handler != null) {
+                handler.handleEvent(playstate, hostname, embyport);
+                logger.trace("Handler was found for thing {}", thing.getLabel());
+            } else {
+                logger.trace("There is no handler for thing {}", thing.getLabel());
+            }
+        });
+    }
+
+    @Override
+    public void updateConnectionState(boolean connected) {
+        if (connected) {
+            updateStatus(ThingStatus.ONLINE);
+            reconnectionCount = 0;
+        } else {
+            reconnectionCount++;
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "The connection to the emby server was closed, binding will wait one min and attempt to restablish connection. There have been "
+                            + Integer.toString(reconnectionCount) + " attempts to establish a new connection");
```

Comment:
Shorter detailed state is better. Maybe something like:
Connection lost. Retrying in 1 minute. Attempt x

Reply:
revised